### PR TITLE
Director detailview

### LINKED ISSUE #123: Add way to edit the CameraTimelines in the directorshot
We currently cannot edit the CameraTimelines in the DirectorShot

### PR DESCRIPTION
**Pull request by mennoo1996**
Fixes #123 
Created the detailview for DirectorShots
This means you can now change padding and cameras for a director shot.

### CODE REVIEW

File: src/main/java/control/DetailViewController.java [hunk 1]
```diff
@@ -24,12 +35,197 @@ public DetailViewController(ControllerManager manager) {
...
+    private void camerasDropdownChangeListener(ListChangeListener.Change c) {
+        System.out.println("CHANGE LISTENER");
+        DirectorShot shot = ((DirectorShot) manager.getActiveShotBlock().getShot());
+        c.next();
+        if (c.wasAdded()) {
+            cameraAddedInDropdown((int) c.getAddedSubList().get(0));
+        } else {
+            cameraDeletedInDropdown((int) c.getRemoved().get(0));
+        }
+        //camerasDropdownUpdateHelper();
```
Comment (Markvanderuit) [inline src/main/java/control/DetailViewController.java]: Dead code. Kill it even more.

File: src/main/java/control/DetailViewController.java [hunk 2]
```diff
@@ -24,12 +35,197 @@ public DetailViewController(ControllerManager manager) {
...
+        DirectorShot dShot = ((DirectorShot) manager.getActiveShotBlock().getShot());
+        Iterator<CameraShot> iterator = dShot.getCameraShots().iterator();
+        CameraShot toRemove = null;
+        while (iterator.hasNext()) {
+            CameraShot shot = iterator.next();
+            if (manager.getTimelineControl()
+                    .getShotBlockForShot(shot).getTimetableNumber() == index) {
+                toRemove = shot;
+                break;
+
```
Comment (Markvanderuit) [inline src/main/java/control/DetailViewController.java]: Randomly random whitespace?

File: src/main/java/control/DetailViewController.java [hunk 3]
```diff
@@ -24,12 +35,197 @@ public DetailViewController(ControllerManager manager) {
...
+        manager.getTimelineControl().initShotBlock(index, shot);
+        manager.setActiveShotBlock(dShotBlock);
+    }
 
     /**
      * Init the begincount handlers.
      */
     private void initBeginCount() {
-        detailView.setBeginCount(0);
+        //detailView.setBeginCount(0);
```
Comment (Markvanderuit) [inline src/main/java/control/DetailViewController.java]: Commented lines.

File: src/main/java/control/DetailViewController.java [hunk 4]
```diff
@@ -24,12 +35,197 @@ public DetailViewController(ControllerManager manager) {
...
+            double newVal = Double.parseDouble(newValue);
+            
+            ((DirectorShotBlock) manager.getActiveShotBlock()).setPaddingBefore(newVal);
+            ((DirectorShot) manager.getActiveShotBlock().getShot()).setFrontShotPadding(newVal);
+            ((DirectorShot) manager.getActiveShotBlock().getShot()).getCameraShots().forEach(e -> {
+                    CameraShotBlock shotBlock = manager.getTimelineControl().getShotBlockForShot(e);
+                    shotBlock.setBeginCount(((DirectorShot) manager.getActiveShotBlock()
+                            .getShot()).getBeginCount() - newVal, true);
+                });
+
```
Comment (bart-de-jonge) [inline src/main/java/control/DetailViewController.java]: You could remove this line

File: src/main/java/control/DetailViewController.java [hunk 5]
```diff
@@ -76,7 +272,7 @@ void beginCountUpdateHelper() {
      * Init the endcuont handlers.
      */
     private void initEndCount() {
-        detailView.setEndCount(0);
+        //detailView.setEndCount(0);
```
Comment (Markvanderuit) [inline src/main/java/control/DetailViewController.java]: Ditto.

File: src/main/java/control/DetailViewController.java [hunk 6]
```diff
@@ -124,7 +320,7 @@ private void endCountUpdateHelper() {
      * Init the description handlers.
      */
     private void initDescription() {
-        detailView.setDescription("");
+        //detailView.setDescription("");
```
Comment (Markvanderuit) [inline src/main/java/control/DetailViewController.java]: Ditto.

File: src/main/java/control/DetailViewController.java [hunk 7]
```diff
@@ -148,7 +344,7 @@ void descriptionTextChangedListener(ObservableValue<? extends String> observable
      * Init the name handler.
      */
     private void initName() {
-        detailView.setName("");
+        //detailView.setName("");
```
Comment (Markvanderuit) [inline src/main/java/control/DetailViewController.java]: Ditto.

File: src/main/java/control/DetailViewController.java [hunk 8]
```diff
@@ -24,12 +35,197 @@ public DetailViewController(ControllerManager manager) {
...
+        manager.getTimelineControl().removeCameraShot(toRemove);
+        dShot.getCameraShots().remove(toRemove);
+        dShot.getTimelineIndices().remove(index);
+    }
+    
+    /**
+     * Method for handling a select in the dropdown.
+     * @param index the index of the camera that was selected.
+     */
+    private void cameraAddedInDropdown(int index) {
```
Comment (bart-de-jonge) [inline src/main/java/control/DetailViewController.java:205]: Maybe add 2 or three whitelines in legit places to improve readability

File: src/main/java/control/DetailViewController.java [hunk 9]
```diff
@@ -173,14 +369,55 @@ void nameTextChangedListener(ObservableValue<? extends String> observable,
      */
     public void activeBlockChanged() {
```
Comment (bart-de-jonge) [inline src/main/java/control/DetailViewController.java:363]: Maybe add some white lines to imporve readability

File: src/main/java/gui/centerarea/DirectorShotBlock.java [hunk 10]
```diff
@@ -15,16 +18,57 @@
...
+                             EventHandler<DirectorShotBlockUpdatedEvent> handler,
+                             DirectorShot shot) {
+        this(shotId, rootCenterArea, beginCount, endCount, description,
+                shot.getName(), handler, shot);
+        
+        this.paddingBefore = paddingBefore;
+        this.paddingAfter = paddingAfter;
+        this.timelineIndices = timelineIndices;
+        
+
```
Comment (Markvanderuit) [inline src/main/java/gui/centerarea/DirectorShotBlock.java]: Double whitespace.

File: src/main/java/gui/headerarea/DirectorDetailView.java [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+     */
+    public DirectorDetailView() {
+        super();
+        directorItemsBox = new HBox();
+        directorItemsBox.setSpacing(TweakingHelper.GENERAL_SPACING);
+        initPaddingBefore();
+        initPaddingAfter();
+        initSelectCameras();
+        this.getChildren().add(directorItemsBox);
+
```
Comment (Markvanderuit) [inline src/main/java/gui/headerarea/DirectorDetailView.java]: Random whitespace.

Comment (bart-de-jonge) [inline src/main/java/gui/headerarea/DirectorDetailView.java]: remove this line

**Comment by Markvanderuit** [comment]
Left some comments concerning whitespace/dead code. Nothing special. Fix that, and then we'll definitely merge.

**Comment by bart-de-jonge** [comment]
Looks fine to me

**Pull request merged**